### PR TITLE
Java instead of shell

### PR DESCRIPTION
**Pull request by srose**

### CODE REVIEW

**Comment by srose** [comment]
Everu sh and cmd should be obsolete. Please lets decide when to merge and remove the .sh and .cmd files.

File: bin/applyKeycloakConfigCli.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+class runKeycloakConfigCli {
```
Comment (thomasdarimont) [inline bin/applyKeycloakConfigCli.java:9]: I think we should rename this to `applyRealmConfig`

File: bin/cli.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Controller script to start the Keycloak environment.
+ *
+ * <h2>Run jboss-cli in Keycloak container</h2>
+ * <pre>{@code
+ *  java bin/cli.java
+ * }</pre>
+ */
+class cli {
```
Comment (thomasdarimont) [inline bin/cli.java:13]: We should just call this `jbossCli`

File: bin/createTlsCerts.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * <h2>Show help</h2>
+ * <pre>{@code
+ *  java createTlsCerts.java --help
+ * }</pre>
+ * <p>
+ * Hint:
+ */
+
+class createTlsCerts {
+    static final String HELP_CMD = "--help";
```
Comment (thomasdarimont) [inline bin/createTlsCerts.java:27]: add newline

File: bin/exportRealm.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        if(additionalOptions != null && !"".equals(additionalOptions.trim())) {
+            deployTriggerCommand.add(additionalOptions);
+        }
+
+        var pb = new ProcessBuilder(deployTriggerCommand);
+        pb.redirectErrorStream(true);
+        var process = pb.start();
+        InputStream stdIn = process.getInputStream();
+        InputStreamReader isr = new InputStreamReader(stdIn);
+        BufferedReader br = new BufferedReader(isr);
```
Comment (thomasdarimont) [inline bin/exportRealm.java:71]: Replace with java.util.Scanner

```
java.util.Scanner

try(var s = new Scanner(inputStream)) {
    while(s.hasNextLine()) {
        String line = s.nextLine();
    }
}
```

File: bin/prerequisites.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            System.out.println("Please install docker-compose.");
+        }
+
+        /*Check directories exist */
+        List<String> requiredDirectories = List.of("./keycloak/extensions/target/classes");
+        requiredDirectories.forEach(requiredDirectoryString ->
+        {
+            var requiredDirectory = new File(requiredDirectoryString);
+            if(!requiredDirectory.exists()) {
+                System.out.printf("Path \"%s\" required. Please create it or run the appropriate tool.%n",requiredDirectoryString);
```
Comment (thomasdarimont) [inline bin/prerequisites.java:38]: Add note about building the project

File: bin/runDefinedKeycloakConfigCli.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        provisioningCommand.add("--network");
+        provisioningCommand.add("host");
+        provisioningCommand.add("--env-file");
+        provisioningCommand.add(envFile);
+        provisioningCommand.add("-e");
+        provisioningCommand.add("IMPORT_PATH=" + fileOrDirectoryNameOfConfig);
+        provisioningCommand.add("-v");
+        provisioningCommand.add(pathToConfig + ":" + CONFIG_PATH_IN_CONTAINER);
+
+        provisioningCommand.add("adorsys/keycloak-config-cli:latest");
```
Comment (thomasdarimont) [inline bin/runDefinedKeycloakConfigCli.java:71]: make version configurable

File: bin/runDefinedKeycloakConfigCli.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+class runDefinedKeycloakConfigCli {
```
Comment (thomasdarimont) [inline bin/runDefinedKeycloakConfigCli.java:12]: rename to keycloakConfigCli

File: bin/runDefinedKeycloakConfigCli.env [hunk 8]
```diff
@@ -0,0 +1,8 @@
+KEYCLOAK_URL=https://id.acme.test:8443/auth
```
Comment (thomasdarimont) [inline bin/runDefinedKeycloakConfigCli.env:1]: rename file to keycloakConfigCli.default.env

File: bin/scanImage.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+class scanImage {
+    static final String HELP_CMD = "--help";
+
+    static final String IMAGE_NAME_OPT = "--image-name";
+    static final String IMAGE_NAME_ENV = "IMAGE_NAME";
+    static final String IMAGE_NAME_DEFAULT = "acme/acme-keycloak:13.0.1.0.0.1.0-SNAPSHOT";
```
Comment (thomasdarimont) [inline bin/scanImage.java:11]: remove default image, make this a mandatory parameter

File: config/stage/dev/realms/foldertest/acme-test.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+realm: acme-test
```
Comment (thomasdarimont) [inline config/stage/dev/realms/foldertest/acme-test.yaml:1]: remove foldertest

File: stop.java [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Controller script to stop the Keycloak environment.
+ *
+ * <h2>Stop Keycloak</h2>
+ * <pre>{@code
+ *  java stop.java
+ * }</pre>
+ */
+class start {
```
Comment (thomasdarimont) [inline stop.java:13]: rename to stop

**Pull request closed without merging**